pkg/controller/command/verifiable/command.go

```diff
@@ -303,32 +385,207 @@ func (o *Command) GeneratePresentationByID(rw io.Writer, req io.Reader) command.
+
+		return command.NewValidationError(GeneratePresentationErrorCode,
+			fmt.Errorf("failed to get did doc from store : %w", err))
+	}
+
+	return o.generatePresentationByID(rw, vc, doc)
 }
 
-func (o *Command) generatePresentation(rw io.Writer, vc *verifiable.Credential,
-	commandMethodName string, errCode command.Code) command.Error {
-	vp, err := vc.Presentation()
+func (o *Command) generatePresentation(rw io.Writer, vcs []*verifiable.Credential, pk string) command.Error {
+	// prepare vp
+	vp, err := o.createAndSignPresentation(vcs, pk)
 	if err != nil {
-		logutil.LogError(logger, commandName, commandMethodName, "generate vp : "+err.Error())
+		logutil.LogError(logger, commandName, generatePresentationCommandMethod, "prepare vp: "+err.Error())
 
-		return command.NewValidationError(errCode, fmt.Errorf("generate vp : %w", err))
+		return command.NewValidationError(GeneratePresentationByIDErrorCode, fmt.Errorf("prepare vp: %w", err))
 	}
 
-	// TODO https://github.com/hyperledger/aries-framework-go/issues/1422 - add proof
+	return o.generate(rw, vp)
+}
 
+func (o *Command) generatePresentationByID(rw io.Writer, vc *verifiable.Credential, didDoc *did.Doc) command.Error {
+	// prepare vp by id
+	vp, err := o.createAndSignPresentationByID(vc, didDoc)
+	if err != nil {
+		logutil.LogError(logger, commandName, generatePresentationCommandMethod, "prepare vp by id: "+err.Error())
+
+		return command.NewValidationError(GeneratePresentationByIDErrorCode, fmt.Errorf("prepare vp by id: %w", err))
+	}
+
+	return o.generate(rw, vp)
+}
+
+func (o *Command) generate(rw io.Writer, vp *verifiable.Presentation) command.Error { // nolint: interfacer
 	vpBytes, err := vp.MarshalJSON()
 	if err != nil {
-		logutil.LogError(logger, commandName, commandMethodName, "generate vp : "+err.Error())
+		logutil.LogError(logger, commandName, generatePresentationCommandMethod, "generate vp: "+err.Error())
 
-		return command.NewValidationError(errCode, fmt.Errorf("generate vp : %w", err))
+		return command.NewValidationError(GeneratePresentationByIDErrorCode, fmt.Errorf("generate vp : %w", err))
 	}
 
 	command.WriteNillableResponse(rw, &Presentation{
 		VerifiablePresentation: string(vpBytes),
 	}, logger)
 
-	logutil.LogDebug(logger, commandName, commandMethodName, "success")
+	logutil.LogDebug(logger, commandName, generatePresentationCommandMethod, "success")
 
 	return nil
 }
+
+func (o *Command) createAndSignPresentation(credentials []*verifiable.Credential,
+	pk string) (*verifiable.Presentation, error) {
+	var err error
+
+	var vp *verifiable.Presentation
+
+	var vcs []interface{}
+	// loop through all credentials present in the request and create unsigned presentation
+	for i, vc := range credentials {
+		if i == 0 {
+			vp, err = vc.Presentation()
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		vcs = append([]interface{}{vc}, vcs...)
+	}
+
+	// Add array of credentials in the presentation
+	err = vp.SetCredentials(vcs...)
```

Comment:
`vp` looks to be nil if `i != 0`.

Reply:
if len(request.VerifiableCredentials) == 0 {
		return nil, "", fmt.Errorf("no credential found")
	}
we have a parsing check in the beginning which make. sure. that credentials in the request exists.

Reply:
Here we are picking the first credential to create the presentation object. Validation has been done prior to that.